helm/adelphi/templates/cassandra-image-builder.yaml

```diff
@@ -0,0 +1,69 @@
+apiVersion: argoproj.io/v1alpha1
+kind: WorkflowTemplate
+metadata:
+  name: cassandra-image
+  namespace: {{ .Values.namespace }}
+  annotations:
+    "helm.sh/hook": pre-install
+spec:
+  templates:
+  - name: cassandra-image-template
+    inputs:
+      parameters:
+      - name: registry_ip
+      - name: commit_hash
+    container:
+      image: gcr.io/kaniko-project/executor:v1.0.0
+      args:
+      - --dockerfile=/dockerfile/Dockerfile
+      - --context=dir:///workspace
+      - {{`--destination={{inputs.parameters.registry_ip}}:30000/cassandra-quality/cassandra`}}
+      - --insecure=true
+      volumeMounts:
+        - name: workspace
+          mountPath: /workspace
+        - name: dockerfile
+          mountPath: /dockerfile
+    initContainers:
+    - name: clone-cassandra
+      image: alpine/git
+      workingDir: /download
+      command:
+        - sh
+        - -c
+      args:
+        - git clone https://github.com/apache/cassandra.git cassandra;
+          cd cassandra;
+          git checkout {{`{{inputs.parameters.commit_hash}}`}};
```

Comment:
Nit: in the case of, say, "cassandra-4.0-beta2" (which looks to be the default given values.yaml) this represents a tag rather than a commit hash.  "tree-ish" is probably the right name here (since that's ultimately what's being passed to "git checkout") but that name... isn't great.  This name could be a branch, it could be a tag, or it could indeed be a commit.  Maybe... "git_identifier" or something?

That said, the term "tree-ish" is reasonably well understood, at least within the git community.  So maybe we just go with that... ?

I guess I lean a bit more towards "git_identifier"... but I could be swayed either way.

Reply:
@absurdfarce I gave this some more thought as well, I wasn't really happy with "hash" either for the same reasons you mentioned.
Tree-ish is a bit confusing to me actually (it makes me think of a path) even though that's what I see in the "git checkout" command help. Strictly speaking, it seems like "commit" and "tree" are two different object types in git: https://matthew-brett.github.io/curious-git/git_object_types.html.

Having said that, I'm also leaning towards "git_identifier" or maybe even "git_checkout" to be self-explanatory (either way, this will be documented).
Let me know what you think.

Reply:
I do think "git_identifier" is the way to go here.  We're talking about something in the git log so "checkout" doesn't really seem right since that's not really what the user is specifying with that value.  That value specifies a tag or a specific commit, something like that... and "identifier" seems general enough to account for all of those options.